### PR TITLE
Add fetching validators by indices and public keys

### PR DESCRIPTION
**Pull request by 0xKiwi**
This PR adds the support for the `ListValidators` endpoint to retrieve validators by indice and public key.

### CODE REVIEW

File: beacon-chain/rpc/beacon/validators.go [hunk 1]
```diff
@@ -199,16 +199,54 @@ func (bs *Server) ListValidators(
 	}
 
 	validatorList := make([]*ethpb.Validators_ValidatorContainer, 0)
-	for i := 0; i < headState.NumValidators(); i++ {
-		val, err := headState.ValidatorAtIndex(uint64(i))
+
+	for _, indice := range req.Indices {
```
Comment (rauljordan) [inline beacon-chain/rpc/beacon/validators.go]:
```suggestion
	for _, idx := range req.Indices {
```

Comment (rauljordan) [reply beacon-chain/rpc/beacon/validators.go]: or maybe valIdx

Comment (terencechain) [inline beacon-chain/rpc/beacon/validators.go]:
```suggestion
	for _, index := range req.Indices {
```

I think index is more popular among our codebase so let's stick with that convention

File: beacon-chain/rpc/beacon/validators.go [hunk 2]
```diff
@@ -124,6 +121,9 @@ func (bs *Server) ListValidatorBalances(
 		}
 		balancesCount = len(res)
 	}
+	sort.Slice(res, func(i, j int) bool {
```
Comment (terencechain) [inline beacon-chain/rpc/beacon/validators.go]: why do we need to sort here?

Comment (0xKiwi) [reply beacon-chain/rpc/beacon/validators.go]: Results are normally sorted if not using indices and public keys, figured it should be sorted for other requests as well. Wdyt?

Comment (terencechain) [reply beacon-chain/rpc/beacon/validators.go]: that sounds good to me!

**Pull request merged**